app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -214,79 +207,42 @@ public void onNotificationPosted(final StatusBarNotification sbn) {
             if (lastNotificationTime != null
                     && currentTimeMillis < lastNotificationTime + mNotifLimitDurationMillis) {
                 return;
-            } else {
-                mLastNotificationTimeMap.put(appPackageName, currentTimeMillis);
             }
+            mLastNotificationTimeMap.put(appPackageName, currentTimeMillis);
         }
 
-        NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
-        String notificationText, notificationBigText, filterText;
+        String filter;
         boolean discardEmptyNotifications;
 
-        {
-            AppSelection appSelection = AppSelectionsStore.get(this).getAppSelection(appPackageName);
-            if (appSelection == null) { // Should never happen. So if it does, just return false
-                filterText = "";
-                discardEmptyNotifications = false;
-            } else {
-                filterText = appSelection.getFilterText().trim();
-                discardEmptyNotifications = appSelection.isDiscardEmptyNotifications();
-            }
+        AppSelection appSelection = AppSelectionsStore.get(this).getAppSelection(appPackageName);
+        if (appSelection == null) { // Should never happen. So if it does, just return false
+            filter = "";
+            discardEmptyNotifications = false;
+        } else {
+            filter = appSelection.getFilterText().trim();
+            discardEmptyNotifications = appSelection.isDiscardEmptyNotifications();
```

Comment:
This method runs for like what, 5 ms? And doesn't do any heavy allocations.
I don't think any special "GC management" is needed. But ok, I'll restore the block.